src/Tasks/Delete.cs

```diff
@@ -60,43 +93,58 @@ public void Cancel()
         /// </summary>
         public override bool Execute()
         {
+            if (!ValidateInputs())
+            {
+                return false;
+            }
             var deletedFilesList = new List<ITaskItem>();
             var deletedFilesSet = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
```

Comment:
We have a weird mix of things that are case-insensitive always and things that care whether your file system is case sensitive. It might be better to use FileUtilities.PathComparer.